Network/Haskoin/Node/Mempool.hs

```diff
@@ -154,6 +166,29 @@ processTxInv pid tids = do
         -- Publish the transaction download job with average priority
         sendManager $ PublishJob (JobDwnTxs toDwn) (ThisPeer pid) 5
 
+processGetTx :: (MonadIO m, MonadLogger m)
+             => PeerId
+             -> TxHash
+             -> StateT MempoolSession m ()
+processGetTx pid tid = do
+    $(logDebug) $ format $ unwords
+        [ "Requesting transaction", encodeTxHashLE tid, "from wallet." ]
+    modify $
+        \s -> s{ txPeerMap = insertWith (flip (++)) tid [pid] (txPeerMap s) }
```

Comment:
What if the same PeerId sends us the same TxHash twice ? Should we nub the list ?